out_test.go

```diff
@@ -405,13 +405,183 @@ var _ = DescribeTable("pushing semver tags",
 			Error: "no tag specified",
 		},
 	),
+	Entry("bumping aliases with no existing tags",
+		SemverTagPushExample{
+			Tags: []string{},
+
+			Variant:     "",
+			Version:     "1.2.3",
+			BumpAliases: true,
+
+			PushedTags: []string{"1.2.3", "1.2", "1", "latest"},
+		},
+	),
+	Entry("not bumping aliases if a prerelease is given",
+		SemverTagPushExample{
+			Tags: []string{},
+
+			Variant:     "",
+			Version:     "1.2.3-alpha.1",
+			BumpAliases: true,
+
+			PushedTags: []string{"1.2.3-alpha.1"},
+		},
+	),
+	Entry("bumping aliases if only older versions exist",
+		SemverTagPushExample{
+			Tags: []string{"1.2.2"},
+
+			Variant:     "",
+			Version:     "1.2.3",
+			BumpAliases: true,
+
+			PushedTags: []string{"1.2.3", "1.2", "1", "latest"},
+		},
+	),
+	Entry("not bumping anything if a newer patch already exists",
+		SemverTagPushExample{
+			Tags: []string{"1.2.4"},
+
+			Variant:     "",
+			Version:     "1.2.3",
+			BumpAliases: true,
+
+			PushedTags: []string{"1.2.3"},
+		},
+	),
+	Entry("not bumping major if a newer minor already exists",
+		SemverTagPushExample{
+			Tags: []string{"1.3.0"},
+
+			Variant:     "",
+			Version:     "1.2.3",
+			BumpAliases: true,
+
+			PushedTags: []string{"1.2.3", "1.2"},
+		},
+	),
+	Entry("bumping minor and major, but not latest, if a newer major version exists",
+		SemverTagPushExample{
+			Tags: []string{"2.0.0"},
+
+			Variant:     "",
+			Version:     "1.2.3",
+			BumpAliases: true,
+
+			PushedTags: []string{"1.2.3", "1.2", "1"},
+		},
+	),
+	Entry("not bumping major if a newer non-variant minor exists",
+		// rationale: tags are representing versions of a cohesive product, so
+		// its versions should be comparable across variants.
+		SemverTagPushExample{
+			Tags: []string{"1.3.0"},
+
+			Variant:     "foo",
+			Version:     "1.2.3",
+			BumpAliases: true,
+
+			PushedTags: []string{"1.2.3-foo", "1.2-foo"},
```

Comment:
Just for sanity
```suggestion
			PushedTags: []string{"1.2.3-foo", "1.2-foo", "foo"},
```

Reply:
Not sure what you mean here - this would be counter to what the test is saying. We *don't* want `foo` bumped because 1.2.3 is not the latest version of the overall product; 1.3.0 is. Do you think this should behave differently?

Reply:
Oo, good point - I hadn't thought of a situation where you'd want some variants to be intentionally older. I was only thinking of the `alpine`/`ubuntu` case where they represent different 'flavors' of the same product.

I'll change it to the proposed behavior. 👍